### PR TITLE
Update `SET`, `schedule` and `tag` APIs in the C target

### PR DESCRIPTION
**Pull request by housengw**
Please refer to issue #1108 for the names of the APIs before and after the change.

### CODE REVIEW

File: test/C/src/SetWithCustomDestructor.lf
Comment (edwardalee) [inline test/C/src/SetWithCustomDestructor.lf:29]:
```suggestion
        complex_t* val = (complex_t*) malloc(sizeof(complex_t));
```

File: test/C/src/SetWithCustomDestructor.lf
Comment (edwardalee) [inline test/C/src/SetWithCustomDestructor.lf:30]:
```suggestion
        val->size = 10;
```
(No need to make a test more expensive to run).

**Review by edwardalee — CHANGES_REQUESTED** [review]
Small error detected. Approve modulo changing function names to lower case.

**Comment by housengw** [comment]
Seems like a Windows specific problem is keeping `SetWithCustomConstructor` from passing on Windows

**Review by Soroosh129 — APPROVED** [review]
LGTM!

Before merging this, let's make sure everyone is aware of this change (maybe by announcing it on lf-interest) and that the website and/or the wiki are up-to-date.

**Comment by housengw** [comment]
I can make an announcement. With regards to the website, I think we should merge the PR first. Otherwise we would have documentation that refer to code not in master

**Review by edwardalee — APPROVED** [review]
LGTM!  We can immediately update the website repo because that website is not live. But anyway, I think we should forge ahead with this...

**Pull request merged**
